dask/dataframe/shuffle.py

```diff
@@ -845,6 +823,9 @@ def collect(p, part, meta, barrier_token):
 
 
 def set_partitions_pre(s, divisions, ascending=True, na_position="last"):
+    if pd.api.types.is_dtype_equal(divisions.dtype, "string"):
+        # with type "string" and pd.NA values, searchsorted will fail
+        divisions = divisions.astype(object)
```

Comment:
It's not totally clear to me how this is fixing the `pd.NA` issue here. IIUC converting to `object` will still result in `pd.NA`s being in `divisions`

Reply:
I know, it looks very hacky, but it works:

```python
# this fails: ValueError: searchsorted requires array to be sorted, which is impossible with NAs present
pd.Series([None, None], dtype='string[pyarrow]').searchsorted(pd.Series([], dtype="string[pyarrow]"), side="right")

# this works
In [9]: pd.Series([None, None]).searchsorted(pd.Series([], dtype="string[pyarrow]"), side="right")
Out[9]: array([], dtype=int64)

In [10]: pd.Series([None, None], dtype='string[pyarrow]').searchsorted(pd.Series([], dtype="string[pyarrow]"), side="right")
```